### PR TITLE
Sb 01

### PR DESCRIPTION
**Pull request by Alex-Korzun**

### CODE REVIEW

**Review by vlykhvar — APPROVED** [review]

File: src/main/java/com/banking/bankservice/repository/RoleRepository.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.banking.bankservice.repository;
+
+import com.banking.bankservice.model.Role;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+public interface RoleRepository extends JpaRepository<Role, Long> {
+    @Query("SELECT r FROM roles r WHERE r.roleName = ?1")
+    Optional<Role> getByName(String name);
```
Comment (KseniiaMakarova) [inline src/main/java/com/banking/bankservice/repository/RoleRepository.java]:
```suggestion
    Optional<Role> getByRoleName(Role.RoleName name);
```
Spring Data JPA is capable of automatically build the query in many cases based on the method name and parameters)

File: src/main/java/com/banking/bankservice/service/UserService.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.banking.bankservice.service;
+
+import com.banking.bankservice.model.User;
+import java.util.Optional;
+
+public interface UserService {
+    User create(User user);
+
+    User update(User user);
```
Comment (KseniiaMakarova) [inline src/main/java/com/banking/bankservice/service/UserService.java]: see checklist

File: src/main/java/com/banking/bankservice/service/UserService.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+    Optional<User> getById(Long id);
+
+    Optional<User> getByPhone(String phone);
```
Comment (KseniiaMakarova) [inline src/main/java/com/banking/bankservice/service/UserService.java]: let's return entity from the service layer

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
let's configure access to h2 database through the console in `application.properties`

File: src/main/java/com/banking/bankservice/repository/RoleRepository.java (hunk 1 above)
Comment (Alex-Korzun) [reply src/main/java/com/banking/bankservice/repository/RoleRepository.java]: done

File: src/main/java/com/banking/bankservice/service/UserService.java (hunk 2 above)
Comment (Alex-Korzun) [reply src/main/java/com/banking/bankservice/service/UserService.java]: refactored and made only one method `save()`

File: src/main/java/com/banking/bankservice/service/UserService.java (hunk 3 above)
Comment (Alex-Korzun) [reply src/main/java/com/banking/bankservice/service/UserService.java]: refactored

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**